### ISSUE TITLE
Send out reminders before scheduled stream time

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#2373]
### Share your bug report, feature request, or comment.

Via ActivityPub.

**Comment by gabek** [comment]
Would love to be able to send out an `Event` object along with this, but Mastodon will ignore the entire activity if there are multiple objects. And unfortunately, we have to be "Mastodon compatible". I hate this.

https://github.com/mastodon/mastodon/issues/19111

**Comment by germainelee** [comment]
Regarding the reminders, I'm thinking of giving them the option to customise the message with provided variables. And option to disable the auto-remind in case they prefer not to.

![image](https://github.com/user-attachments/assets/620ae378-a3ab-41d6-87fe-c4b905b81b16)

**Comment by gabek** [comment]
I'd prefer this to work the same way that automated messages already work, and we don't currently support this. Starting simpler is always better, and it can be iterated on if necessary, but the current solution seems to be working.

**Comment by germainelee** [comment]
I don't understand. So there's a Now Live message text field for user to enter custom message, but there won't be a text field to enter reminder message?

**Comment by gabek** [comment]
You do set a live message, but you don't set all the additional data (url, server name, etc). That gets set for you.

**Comment by germainelee** [comment]
oic. I'm guessing the post would look like this with the now live message in the configure social features?

> This is a test stream on a test server. You can drop by if you like, but it won't be much fun. 🤗
> 
> `{stream title if have}`
>
> #owncast #streaming
> 
> [nightly.owncast.tv/](https://nightly.owncast.tv/)

![image](https://github.com/user-attachments/assets/0c86c50c-080a-4863-9d72-11376bac5ad9)

**Comment by gabek** [comment]
Yup! Here's an example. You can search the fediverse for #owncast and see a bunch of them.

![image](https://github.com/user-attachments/assets/ff89e8f1-f749-473b-b95e-0590b5e2da60)

**Comment by germainelee** [comment]
As the formatting will be set for you, it would be nice if they could preview it or some kind of text to inform them how the reminder post would look like.
